service/src/config.rs

```diff
@@ -6,45 +6,87 @@ pub struct Config {
 	pub node_ip: String,
 	pub node_port: String,
 	pub worker_ip: String,
-	pub worker_rpc_port: String,
-	/// listening port for the mutual-remote attestation requests
-	pub worker_mu_ra_port: String,
-	/// Todo: Is this deprecated? I can only see it in `enclave_perform_ra`
-	pub ext_api_url: Option<String>,
+	/// Worker address that will be advertised on the parentchain. Should be used when the worker is running
+	/// behind an nginx or docker server.
+	pub external_worker_address: Option<String>,
+	/// Port to directly communicate with the trusted tls server inside the enclave.
+	pub trusted_worker_port: String,
+	/// Port to the untrusted ws of the validateer.
+	pub untrusted_worker_port: String,
+	/// Port for mutual-remote attestation requests.
+	pub mu_ra_port: String,
 }
 
 impl Config {
 	pub fn new(
 		node_ip: String,
 		node_port: String,
 		worker_ip: String,
-		worker_rpc_port: String,
-		worker_mu_ra_port: String,
+		external_worker_address: Option<String>,
+		trusted_worker_port: String,
+		untrusted_worker_port: String,
+		mu_ra_port: String,
 	) -> Self {
 		Self {
 			node_ip,
 			node_port,
 			worker_ip,
-			worker_rpc_port,
-			worker_mu_ra_port,
-			ext_api_url: None,
+			external_worker_address,
+			trusted_worker_port,
+			untrusted_worker_port,
+			mu_ra_port,
 		}
 	}
 
+	/// Returns the client url of the node (including ws://).
 	pub fn node_url(&self) -> String {
 		format!("{}:{}", self.node_ip, self.node_port)
 	}
 
-	pub fn worker_url(&self) -> String {
-		format!("{}:{}", self.worker_ip, self.worker_rpc_port)
+	pub fn trusted_worker_url_internal(&self) -> String {
+		format!("{}:{}", self.worker_ip, self.trusted_worker_port)
+	}
+
+	/// Returns the trusted worker url that should be addressed by external clients.
+	///
+	/// In case external_address is set, it should be considered that the internal trusted server is a tls
+	/// websocket and must have have a wss:// primary to the internal worker ip.
+	pub fn trusted_worker_url_external(&self) -> String {
+		match &self.external_worker_address {
+			Some(external_address) => format!("{}:{}", external_address, self.trusted_worker_port),
+			None => format!("wss://{}:{}", self.worker_ip, self.trusted_worker_port),
+		}
+	}
+
+	pub fn untrusted_worker_url(&self) -> String {
+		format!("{}:{}", self.worker_ip, self.untrusted_worker_port)
+	}
+
+	/// Returns the untrusted worker url that should be addressed by external clients.
+	///
+	/// In case external_address is set, it should be considered that the internal untrusted worker url
+	/// must have a ws:// primary to the internal worker ip.
+	pub fn untrusted_worker_url_external(&self) -> String {
+		match &self.external_worker_address {
+			Some(external_address) =>
+				format!("{}:{}", external_address, self.untrusted_worker_port),
+			None => format!("ws://{}:{}", self.worker_ip, self.untrusted_worker_port),
+		}
 	}
```

Comment:
Hmm, this one is very tricky because we will always have wss in the published external address, but it will not always be wss for the untrusted part.

Spontaneously, I have no idea how to solve that cleanly such that it always works.

Reply:
Me neither. Added all external urls as an input option. I think we should have covered all cases like that. Maybe not in an elegant way, but it should work for all cases.